elfloader-net/README.md

```diff
@@ -0,0 +1,288 @@
+# Unikraft ELF Loader (Networking)
+
+Build and run the [Unikraft ELF Loader](https://github.com/unikraft/app-elfloader) with networking support.
+The ELF Loader uses the [Unikraft binary-compatibility layer](https://unikraft.org/docs/concepts/compatibility) in order to run native Linux binaries (ELFs) with Unikraft.
+The ELF has to be PIE (*Position-Independent Executable*), either static or dynamic.
+
+Follow the instructions below to set up, configure, build and run the ELF Loader.
+Make sure you installed the [requirements](../README.md#requirements).
+
+The Unikraft ELF Loader currently runs on x86_64 (ARM64 work is underway) and on KVM (QEMU and Firecracker) (Xen work is underway).
+
+## Quick Setup (aka TLDR)
+
+For a quick setup, run the commands below.
+Note that you still need to install the [requirements](../README.md#requirements).
+
+To build and run a simple HTTP server Linux ELF using the Unikraft ELF Loader, use the commands below:
+
+```console
+test -d ../repos/unikraft || git clone https://github.com/unikraft/unikraft ../repos/unikraft
+test -d ../repos/apps/elfloader || git clone https://github.com/unikraft/app-elfloader ../repos/apps/elfloader
+test -d ../repos/libs/libelf || git clone https://github.com/unikraft/lib-libelf ../repos/libs/libelf
+test -d "../repos/libs/lwip" || git clone https://github.com/unikraft/lib-lwip ../repos/libs/lwip
+make distclean
+> /tmp/defconfig echo 'CONFIG_PLAT_KVM=y
+CONFIG_KVM_VMM_QEMU=y
+CONFIG_ARCH_X86_64=y
+CONFIG_APPELFLOADER_DEPENDENCIES=y
+CONFIG_APPELFLOADER_ARCH_PRCTL=y
+CONFIG_APPELFLOADER_BRK=y
+CONFIG_APPELFLOADER_CUSTOMAPPNAME=y
+CONFIG_APPELFLOADER_STACK_NBPAGES=128
+CONFIG_APPELFLOADER_VFSEXEC_EXECBIT=n
+CONFIG_APPELFLOADER_VFSEXEC=y
+CONFIG_APPELFLOADER_HFS=y
+CONFIG_APPELFLOADER_HFS_ETCRESOLVCONF=y
+CONFIG_APPELFLOADER_HFS_ETCHOSTS=y
+CONFIG_APPELFLOADER_HFS_ETCHOSTNAME=y
+CONFIG_APPELFLOADER_HFS_REPLACEEXIST=y
+CONFIG_LIBVFSCORE=y
+CONFIG_LIBVFSCORE_AUTOMOUNT_UP=y
+CONFIG_LIBRAMFS=y
+CONFIG_LIBUKCPIO=y
+CONFIG_LIBDEVFS=y
+CONFIG_LIBDEVFS_AUTOMOUNT=y
+CONFIG_LIBLWIP=y
+CONFIG_LIBUKNETDEV_EINFO_LIBPARAM=y'
+UK_DEFCONFIG=/tmp/defconfig make defconfig
+make -j $(nproc)
+make -C rootfs/
+test -f initrd.cpio || ../repos/unikraft/support/scripts/mkcpio initrd.cpio ./rootfs/
```

Comment:
Should we add the network interface setup here as well? The `ip link add...` commands?

Reply:
> Should we add the network interface setup here as well? The `ip link add...` commands?

solved, thanks